codersdk/agentsdk/agentsdk.go

```diff
@@ -206,14 +206,11 @@ func (c *Client) ConnectRPC(ctx context.Context) (drpc.Conn, error) {
 		return nil, codersdk.ReadBodyAsError(res)
 	}
 
-	_, wsNetConn := codersdk.WebsocketNetConn(ctx, conn, websocket.MessageBinary)
+	// Set the read limit to 4 MiB -- about the limit for protobufs.  This needs to be larger than
+	// the default because some of our protocols can include large messages like startup scripts.
+	conn.SetReadLimit(1 << 22)
+	netConn := websocket.NetConn(ctx, conn, websocket.MessageBinary)
```

Comment:
Since this is no longer wrapped via custom closer below, there's a slight change in behavior, this will be used instead:

```go
c.c.Close(StatusNormalClosure, "")
```

Reply:
True, but I don't think it matters in any meaningful way.  The server doesn't act any differently based on the close code or message.